### ISSUE TITLE
python notification not running when icinga ran as a service

### ISSUE THREAD

**Issue by pdenning** [Icinga/icinga2#6028]
I have hipsaint python package installed to forward notifications into our onsite hipchat server. When a notification is sent. It doesn't get forwarded to the hipchat server.

When running the daemon from the command line with icinga2 daemon -d notifications are sent.

When running as a service via systemctl start icinga2 no notifications are sent and python errors out.

If i set selinux to Permissive mode notifcations are send when running as a service.

No selinux denies in audit.log

## Expected Behavior
The notification to be sent the to hipchat server

## Current Behavior
Notifcation is not set to the hipchat server

The following error is found within /var/log/icinga2/icinga2.log (-x debug) was added to icinga2.service to retrieve the error.

    ```
    [2018-01-23 22:10:29 +0000] information/Checkable: Checking for configured notifications for object 'host.local!memory_used'
    [2018-01-23 22:10:29 +0000] information/Notification: Sending 'Custom' notification 'host.local!memory_used!notify-noc-all-services' for user 'NOC'
    [2018-01-23 22:10:29 +0000] information/Notification: Completed sending 'Custom' notification 'host.local!memory_used!notify-noc-all-services' for checkable 'host.local!memory_used' and user 'soc'.
    [2018-01-23 22:10:29 +0000] warning/PluginNotificationTask: Notification command for object 'host.local!memory_used' (PID: 11637, arguments: '/etc/icinga2/scripts/hipsaint-service-notification.sh' '-h' '<hipchat.server>' '-i' 'memory_used|HOST|2018-01-23 22:10:29 +0000|CUSTOM|host.local|WARNING|WARNING - 25.0% (1012384 kB) used!' '-r' '6' '-t' '<TOKEN REMOVED>') terminated with exit code 1, output: Traceback (most recent call last):
      File "/usr/lib64/python2.7/site.py", line 556, in <module>
        main()
      File "/usr/lib64/python2.7/site.py", line 538, in main
        known_paths = addusersitepackages(known_paths)
      File "/usr/lib64/python2.7/site.py", line 266, in addusersitepackages
        user_site = getusersitepackages()
      File "/usr/lib64/python2.7/site.py", line 241, in getusersitepackages
        user_base = getuserbase() # this will also set USER_BASE
      File "/usr/lib64/python2.7/site.py", line 231, in getuserbase
        USER_BASE = get_config_var('userbase')
      File "/usr/lib64/python2.7/sysconfig.py", line 516, in get_config_var
        return get_config_vars().get(name)
      File "/usr/lib64/python2.7/sysconfig.py", line 473, in get_config_vars
        _CONFIG_VARS['userbase'] = _getuserbase()
      File "/usr/lib64/python2.7/sysconfig.py", line 187, in _getuserbase
        return env_base if env_base else joinuser("~", ".local")
      File "/usr/lib64/python2.7/sysconfig.py", line 173, in joinuser
        return os.path.expanduser(os.path.join(*args))
      File "/usr/lib64/python2.7/posixpath.py", line 269, in expanduser
        userhome = pwd.getpwuid(os.getuid()).pw_dir
    KeyError: 'getpwuid(): uid not found: 997'
    ```

disabling selinux corrected the issue.

## Possible Solution
changes to selinux to allow python packages to access the icinga2 home directory (/var/spool/icinga2) or whatever black magic selinux needs.

## Steps to Reproduce (for bugs)
1. sent selinux to enforce
2. install hipsaint  python module (pip install hipsaint)
3. create hipsaint shell script

    ```
    #!/bin/bash
    
    NOTIFY=False
    
    while [[ $1 = -* ]]; do
      arg=$1; shift
    
      case $arg in
        -r) ROOM_ID=$1
            shift
            ;;
        -t) AUTH_TOKEN=$1
            shift
            ;;
        -h) HIPCHAT_SERVER=$1;
            shift
            ;;
        -n) NOTIFY=True
            ;;
        -i) INPUTS=$1;
            shift
            ;;
      esac
    done
    
    hipsaint --host $HIPCHAT_SERVER -V 2 --token $AUTH_TOKEN --room $ROOM_ID --type "service" --inputs "$INPUTS" -n -s
    ``` 

3. create hipsaint cmd
    ```
    template NotificationCommand "hipsaint-service-notification-tpl" {
        import "plugin-notification-command"
        command = [ "/etc/icinga2/scripts/hipsaint-service-notification.sh" ]
        arguments += {
            "-h" = "$hipchat_host$"
            "-i" = "$hipsaint_inputs$"
            "-r" = "$hipchat_room_id$"
            "-t" = "$hipchat_token$"
        }
    }
    ```
4. add notifcation for hipsaint
    ```
    template Notification "hipchat-service-alarm-tpl" {
        command = "hipchat-service-cmd"
        states = [ Critical, OK, Unknown, Warning ]
        types = [ Acknowledgement, Custom, Problem, Recovery ]
        users = [ "noc" ]
    }
    ```
    
    ```
    apply Notification "notify-soc-all-services" to Service {
        import "hipsaint-service-notify-tpl"
    
        assign where service.notify_hipchat
        users = [ "noc" ]
    }
    ```
5. send notification

## Context
Trying to have active alerts sent to a change server allows operators not to have to monitor dashboards 100% of the time.

## Your Environment

* Version used (`icinga2 --version`):
    Icinga2 - The Icinga 2 network monitoring daemon (version: r2.8.1-1)
    
    Copyright (c) 2012-2017 Icinga Development Team (https://www.icinga.com/)
    License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
    This is free software: you are free to change and redistribute it.
    There is NO WARRANTY, to the extent permitted by law.
    
    Application information:
      Installation root: /usr
      Sysconf directory: /etc
      Run directory: /run
      Local state directory: /var
      Package data directory: /usr/share/icinga2
      State path: /var/lib/icinga2/icinga2.state
      Modified attributes path: /var/lib/icinga2/modified-attributes.conf
      Objects path: /var/cache/icinga2/icinga2.debug
      Vars path: /var/cache/icinga2/icinga2.vars
      PID path: /run/icinga2/icinga2.pid
    
    System information:
      Platform: CentOS Linux
      Platform version: 7 (Core)
      Kernel: Linux
      Kernel version: 3.10.0-693.11.1.el7.x86_64
      Architecture: x86_64
    
    Build information:
      Compiler: GNU 4.8.5
      Build host: unknown

* Operating System and version:
CentOS 7 - Selinux enforeced

* Enabled features (`icinga2 feature list`):
Disabled features: command compatlog elasticsearch gelf graphite livestatus opentsdb perfdata statusdata syslog
Enabled features: api checker debuglog ido-pgsql influxdb mainlog notification

* Icinga Web 2 version and modules (System - About):
    icingaweb2 version: 2.5
    Modules:
    director	1.4.2
    grafana	1.1.7
    monitoring	2.5.0

* Config validation (`icinga2 daemon -C`):
    information/cli: Icinga application loader (version: r2.8.1-1)
    information/cli: Loading configuration file(s).
    information/ConfigItem: Committing config item(s).
    information/ApiListener: My API identity: ac2pvlm202.cbrops.protected
    information/ConfigItem: Instantiated 1 ApiListener.
    information/ConfigItem: Instantiated 5 Zones.
    information/ConfigItem: Instantiated 4 Endpoints.
    information/ConfigItem: Instantiated 2 ApiUsers.
    information/ConfigItem: Instantiated 2 FileLoggers.
    information/ConfigItem: Instantiated 1 UserGroup.
    information/ConfigItem: Instantiated 2 NotificationCommands.
    information/ConfigItem: Instantiated 16 Notifications.
    information/ConfigItem: Instantiated 209 CheckCommands.
    information/ConfigItem: Instantiated 1 IcingaApplication.
    information/ConfigItem: Instantiated 5 Hosts.
    information/ConfigItem: Instantiated 1 HostGroup.
    information/ConfigItem: Instantiated 1 User.
    information/ConfigItem: Instantiated 1 ServiceGroup.
    information/ConfigItem: Instantiated 30 Services.
    information/ConfigItem: Instantiated 1 CheckerComponent.
    information/ConfigItem: Instantiated 1 IdoPgsqlConnection.
    information/ConfigItem: Instantiated 1 InfluxdbWriter.
    information/ConfigItem: Instantiated 1 NotificationComponent.
    information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
    information/cli: Finished validating the configuration file(s).

## further debug info

* ps -efZ | grep icinga2 when ran as a service ( notification doesnt work)

    ```
    system_u:system_r:icinga2_t:s0  icinga   18696     1  0 00:18 ?        00:00:01 /usr/lib64/icinga2/sbin/icinga2 --no-stack-rlimit daemon -d -e /var/log/icinga2/error.log -x debug
    system_u:system_r:icinga2_t:s0  icinga   18701     1  0 00:18 ?        00:00:18 /usr/lib64/icinga2/sbin/icinga2 --no-stack-rlimit daemon -d -e /var/log/icinga2/error.log -x debug
    ```

* ps -efZ | grep icinga2 when ran as a from command line (Which does sent notifications)

    ```
    unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 icinga 1745 1  0 02:47 pts/0 00:00:00 /usr/lib64/icinga2/sbin/icinga2 --no-stack-rlimit daemon -d -e /var/log/icinga2/error.log -x debug
    unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 icinga 1750 1 29 02:47 ? 00:00:03 /usr/lib64/icinga2/sbin/icinga2 --no-stack-rlimit daemon -d -e /var/log/icinga2/error.log -x debug
    ```

* ls -Z /etc/icinga2/scripts/hipsaint-service-notification.sh 
   ```-rwxr-xr-x. root root system_u:object_r:nagios_notification_plugin_exec_t:s0 /etc/icinga2/scripts/hipsaint-service-notification.sh```

* ls -Z /usr/bin/hipsaint 
    ```-rwxr-xr-x. root root system_u:object_r:bin_t:s0       /usr/bin/hipsaint```

**Comment by pdenning** [comment]
Hi dgoetz

Thank you for your assistance

I had to disable dontaudit rules to get an out put for the audit2allow -li /var/log/audit/audit.log

  ```
  #============= icinga2_t ==============
  allow icinga2_t nagios_checkdisk_plugin_t:process { noatsecure rlimitinh siginh };
  allow icinga2_t nagios_notification_plugin_t:process { noatsecure rlimitinh siginh };
  allow icinga2_t nagios_unconfined_plugin_t:process { noatsecure rlimitinh siginh };
  
  #============= nagios_notification_plugin_t ==============
  allow nagios_notification_plugin_t passwd_file_t:file read;
  ```
I have used nagios_unconfined_plugin_exec_t on the shell script within /etc/icinga2/scripts as a work around and it appears to be working.

**Comment by mcktr** [comment]
I change the milestone to 2.9.0 since the fixing PR is already merged and 2.9.0 is assigned to them.
